### DISCUSSION TITLE
How to close the rules regularly？

### DISCUSSION THREAD

**Discussion by userzhangqg** [jertel/elastalert2#1479] [Q&A] [answered]
I hope to automatically close the alarm rules after a period of time. How can I do ？

**Comment by jertel** [comment]
I don't understand "close the alarm rules". Can you provide more context?

**Reply by userzhangqg** [reply]
I expect to automatically disable a rule after a period of time. Is there any way to do this?

**Reply by userzhangqg** [reply]
For example, can i configure a deadline, and disable the rules after reaching that time?

**Reply by jertel** [reply, marked as answer]
There is a `limit_execution` option that allows you to control which days and times the rule is allowed to fire. That's not really what you're asking though. Another option is to create an enhancement that skips any matches that occur after the cutoff. That isn't perfect either since it still means the rule is wasting resources conducting the query.

If you're interested in submitting a pull request for a new `disable_datetime` option then you could consider doing that.

**Reply by userzhangqg** [reply]
I know, thank you.
